File: SoundTrekServer/SoundTrekServer/Models/WikipediaLandmark.cs
namespace SoundTrekServer.Models
{
    public class WikipediaLandmark
    {
        public string PageId { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public double Lat { get; set; }
        public double Lon { get; set; }
        public string? ImageUrl { get; set; }
        public string? ShortDescription { get; set; }
        public List<string> Categories { get; set; } = new();
        public string? WikibaseItem { get; set; }
    }
}

File: SoundTrek/types/WikipediaLandmark.ts
// Landmark type (based on server structure)
export type WikipediaLandmark = {
    title: string;
    pageId: string;
    lat: number;
    lon: number;
    imageUrl?: string; 
    shortDescription?: string;
};

export type LandmarkToCard = {
  pageId: string;
  title: string;
  lat: number;
  lon: number;
  imageUrl: string;
  shortDescription: string;
  categories: {
    categoryId: number;
    categoryName: string;
  }[];
  interests: {
    interestId: number;
    interestName: string;
  }[];
  mainCategories: {
    mainCategoryId: number;
    mainCategoryName: string;
  }[];
};